peps/pep-0750.rst

```diff
@@ -718,88 +561,224 @@ interpolations, are as follows:
 1. Scope lookup is the same as f-strings (lexical scope). This model has been
    shown to work well in practice.
 
-2. Tag functions can ensure that any interpolations are done in a safe fashion,
-   including respecting the context in the target DSL.
+2. Code that processes ``Template`` instances can ensure that any interpolations 
+   are processed in a safe fashion, including respecting the context in the 
+   target DSL.
+
 
 How To Teach This
 =================
 
-Tag strings have several audiences: consumers of tag functions, authors of tag
-functions, and framework authors who provide interesting machinery for tag
-functions.
+Template strings have several audiences: developers using template strings
+and processing libraries, authors of template processing code, and framework
+authors who provide interesting machinery built with template strings.
 
 All three groups can start from an important framing:
 
-- Existing solutions (such as template engines) can do parts of tag strings
-- But tag strings move logic closer to "normal Python"
+- Existing solutions (such as template engines) can do parts of template strings
+- But template strings move logic closer to "normal Python"
 
-Consumers can look at tag strings as starting from f-strings:
+Consumers can look at template strings as starting from f-strings:
 
 - They look familiar
 - Scoping and syntax rules are the same
 
-They first thing they need to absorb: unlike f-strings, the string isn't
-immediately evaluated "in-place". Something else (the tag function) happens.
-That's the second thing to teach: the tag functions do something particular.
-Thus the concept of "domain specific languages" (DSLs). What's extra to
-teach: you need to import the tag function before tagging a string.
+The first thing they need to absorb: unlike f-strings, template string literals
+don't evaluate to strings; rather, they evaluate to a new type, ``Template``.
+``Template`` is a simple type intended to be used by template processing code.
+It's not until you call that code that you get the result you want: typically,
+a string, although processing code can of course return any arbitrary type.
 
-Tag function authors think in terms of making a DSL. They have
-business policies they want to provide in a Python-familiar way. With tag
+Template function authors think in terms of making a DSL. They have
+business policies they want to provide in a Python-familiar way. With template
 functions, Python is going to do much of the pre-processing. This lowers
 the bar for making a DSL.
 
-Tag authors can begin with simple use cases. After authors gain experience, tag strings can be used to add larger
+Template authors can begin with simple use cases. After authors gain experience, template strings can be used to add larger
 patterns: lazy evaluation, intermediate representations, registries, and more.
```

Comment:
Maybe we should remove "lazy evaluation" as neither the template string or the function can really add it as a pattern.

Reply:
Good point. I've updated this to more match the rest of the PEP:

> Template authors can begin with simple use cases. After authors gain experience, 
template strings can be used to add larger patterns, like parsing to intermediate
representations or context-sensitive processing of interpolations.